pkg/apis/pipeline/v1beta1/implicit_params_test.go

```diff
@@ -314,160 +297,53 @@ func TestImplicitParams_Pipeline(t *testing.T) {
 
```

Comment:
Im a bit confused about why the Pipeline spec has an implicit params test - it seems like in the updated version, the implicit params logic is only called from the pipelinerun and taskrun defaulting functions - i would think pipeline.SetDefaults wouldn't have any implicit param logic? (unless the purpose of this test is to ensure that the implicit params logic is not applied?)

one thing that might make this a bit clearer is to have test cases specifically for the new implicit params functions  as well - the current test cases seem to be covering the existing defaulting logic as well so it's a bit hard to tell what is being caused by the implicit params logic and what is caused by the defaulting logic